### PR TITLE
Review demo app

### PR DESCRIPTION
**Pull request by mahmoudjafarinejad**
- Reviewed and refactored the demo app
- Added error handling for different scenarios
- Updated UI and dependencies
- Refined the readme file

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,15 +1,105 @@
...
+As previously mentioned Klarna Payments is supported in the native integration. The main idea behind Klarna Payments is to give merchants the ability to offer Klarna’s payment methods to customers. Because Klarna doesn’t own the entire checkout flow, a check needs to be performed before an order is created. Our goal with this integration is to offer you the same simple development experience as you have on the web. As such, integrating Klarna Payments consists of four steps:
+
+1. Creating a session.
+2. Rendering the payment view.
+3. Authorizing the session.
+4. Creating an order.
+
+![](https://developers.klarna.com/static/in-app/kp-diagram.png)
+
+###Screenshots
```
Comment (adrianpalmquist) [inline README.md]: Add an extra space here `### Screenshots`

File: README.md [hunk 2]
```diff
@@ -1,15 +1,105 @@
-# Klarna In-App SDK for Klarna Payments - Android Example App
+# Klarna In-App SDK - Android Demo App
+![Platforms](https://img.shields.io/badge/platform-android-lightgrey)
+[![GitHub](https://img.shields.io/github/license/klarna/klarna-mobile-sdk)](https://github.com/klarna/klarna-mobile-sdk/blob/master/LICENSE)
```
Comment (adrianpalmquist) [inline README.md]: This one is pointing to the License for the SDK not the demo app repo itself

File: README.md [hunk 3]
```diff
@@ -1,15 +1,105 @@
...
+This section will guide you through submitting a bug report for Klarna In-App SDK - Android Demo App.
+
+Before submitting a bug report, please check that the issue hasn't been reported before. If you find a *Closed* issue that seem to describe an issue that is similar to what you want to report, open a new issue and link to the original issue in the new one. When you have checked that the issue hasn't been reported before **fill out [the required template](https://github.com/klarna/klarna-mobile-sdk/blob/a55293185f73609df831a26ddff5a66033f21e55/.github/ISSUE_TEMPLATE/bug_report.md)** which will help us resolve the issue faster. 
+
+##### Submitting a Bug Report
+Bugs that are submitted are tracked as [GitHub issues](https://guides.github.com/features/issues/). To report a bug, create an issue and use [the template](https://github.com/klarna/klarna-mobile-sdk/blob/a55293185f73609df831a26ddff5a66033f21e55/.github/ISSUE_TEMPLATE/bug_report.md) to provide information about the bug. Explain the problem thoroughly and include any additional information that you think might help the maintainers reproduce the issue. When creating the GitHub issue please make sure that you:
```
Comment (adrianpalmquist) [inline README.md]: We need templates for this repo itself

File: README.md [hunk 4]
```diff
@@ -1,15 +1,105 @@
...
+Before contributing please read through the [Klarna In-App SDK documentation](https://developers.klarna.com/documentation/in-app/).
+
+##### Branching
+Prefix the branch you are going to work on depending on what you are working on (bug fix or feature). Use the following prefixes when creating a new branch:
+
+* **feature/** if the branch contains a new feature, example: `feature/my-shiny-feature`.
+* **bugfix/**  if the branch contains a bug fix, example: `bugfix/my-bug-fix`.
+
+##### Pull Requests
+When creating a PR include as much information as possible about the type of enhancement, whether if it's a bugfix, new functionality or any other change. There's [a template](https://github.com/klarna/klarna-mobile-sdk/blob/a55293185f73609df831a26ddff5a66033f21e55/.github/ISSUE_TEMPLATE/pull-request.md) for you to fill out which will make the review process for the maintainers faster. When creating a PR do it against the `master` branch. The PR should include:
```
Comment (adrianpalmquist) [inline README.md]: Same here, need a template for PR's in this repo

File: README.md [hunk 5]
```diff
@@ -1,15 +1,105 @@
...
+*  [Android](https://htmlpreview.github.io/?https://github.com/klarna/klarna-mobile-sdk/blob/master/docs/android/klarna-mobile-sdk/index.html)
+
+## Contributing
+Thank you for reading this and taking time to contribute to Klarna In-App SDK - Android Demo App! Below is a set of guidelines to help you contribute whether you want to report a bug, come with suggestions or modify code.
+
+### How can I contribute?
+#### Reporting Bugs
+This section will guide you through submitting a bug report for Klarna In-App SDK - Android Demo App.
+
+Before submitting a bug report, please check that the issue hasn't been reported before. If you find a *Closed* issue that seem to describe an issue that is similar to what you want to report, open a new issue and link to the original issue in the new one. When you have checked that the issue hasn't been reported before **fill out [the required template](https://github.com/klarna/klarna-mobile-sdk/blob/a55293185f73609df831a26ddff5a66033f21e55/.github/ISSUE_TEMPLATE/bug_report.md)** which will help us resolve the issue faster.
```
Comment (adrianpalmquist) [inline README.md]: Need a template for bug reports in this repo

File: README.md (hunk 1 above)
Comment (mahmoudjafarinejad) [reply README.md]: @adrianpalmquist done.

File: README.md (hunk 2 above)
Comment (mahmoudjafarinejad) [reply README.md]: @adrianpalmquist done.

File: README.md (hunk 3 above)
Comment (mahmoudjafarinejad) [reply README.md]: @adrianpalmquist done.

File: README.md (hunk 4 above)
Comment (mahmoudjafarinejad) [reply README.md]: @adrianpalmquist done.

File: README.md (hunk 5 above)
Comment (mahmoudjafarinejad) [reply README.md]: @adrianpalmquist done.

**Pull request merged**